src/Internal/Processor.php

```diff
@@ -815,7 +815,7 @@ private function handleLocalInfileRequest($packet)
                 $filePath = \substr($packet, 1);
                 /** @var \Amp\File\Handle $fileHandle */
                 $fileHandle = yield \Amp\File\open($filePath, 'r');
-                while (strlen($chunk = yield $fileHandle->read()) > 0) {
+                while ("" != ($chunk = yield $fileHandle->read())) {
```

Comment:
This needs to be `while (null !== $chunk = yield $fileHandle->read()) { if ($chunk === "") { continue; } ... }`

Reply:
Uh, seriously? when does a file read ever return an empty string?

Reply:
Not sure, some implementation might do that for any reason, e.g. a filesystem reading files directly from Amazon S3 might use some custom stream which might return empty chunks.